RUFAS/EEE/emissions.py

```diff
@@ -283,387 +205,438 @@ def _get_feed_emissions_data(
-        applied_crops = []
-
-        for crop in sorted_crops:
-            crop_planting_date = RufasTime.convert_year_jday_to_date(crop["planting_year"], crop["planting_day"]).date()
-            crop_harvest_date = RufasTime.convert_year_jday_to_date(crop["harvest_year"], crop["harvest_day"]).date()
-            if crop_planting_date <= fertilizer_application_date < crop_harvest_date:
-                applied_crops.append(crop)
-        return applied_crops
-
-    def _soil_data_update(self, filters: dict[str, dict[str, Any]]) -> dict[str, int | Literal[0]]:
-        """
-        This method will update the soil data update according to data extracted using the filters.
+        daily_farmgrown_feed_emissions_and_resources: dict[RUFAS_ID, dict[int, dict[str, float]]],
+        feed_deductions_data: dict[RUFAS_ID, dict[int, float]],
+        all_simulation_days: list[int],
+    ) -> dict[RUFAS_ID, dict[int, dict[str, float]]]:
+        """Calculates daily farmgrown feed emissions and resources used for farmgrown feeds fed to the animals."""
+        result: dict[RUFAS_ID, dict[int, dict[str, float]]] = defaultdict(dict)
+        for feed_id, feed_deductions in feed_deductions_data.items():
+            if feed_id not in daily_farmgrown_feed_emissions_and_resources:
+                continue
+            for simulation_day in all_simulation_days:
+                feed_deduction = feed_deductions.get(simulation_day, 0.0)
+                data_for_feed_id_for_day = daily_farmgrown_feed_emissions_and_resources[feed_id][simulation_day]
+                result[feed_id][simulation_day] = {
+                    "nitrous_oxide_emissions": data_for_feed_id_for_day["nitrous_oxide_emissions"] * feed_deduction,
+                    "ammonia_emissions": data_for_feed_id_for_day["ammonia_emissions"] * feed_deduction,
+                    "fertilizer_N": data_for_feed_id_for_day["fertilizer_N"] * feed_deduction,
+                    "fertilizer_P": data_for_feed_id_for_day["fertilizer_P"] * feed_deduction,
+                    "fertilizer_K": data_for_feed_id_for_day["fertilizer_K"] * feed_deduction,
+                    "manure_N": data_for_feed_id_for_day["manure_N"] * feed_deduction,
+                }
+        return result
 
-        Parameters
-        ----------
-        filters: dict[str, dict[str, Any]]
-            A dictionary of filters to retrieve data from the OM.
+    def _report_daily_farmgrown_feed_fed_emissions_and_resources(
+        self,
+        daily_farmgrown_feed_fed_emissions_and_resources: dict[RUFAS_ID, dict[int, dict[str, float]]],
+    ) -> None:
+        """Reports the emissions and resources for daily farmgrown feeds fed to the animals."""
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._report_daily_farmgrown_feed_fed_emissions_and_resources.__name__,
+        }
+        for feed_id, daily_data_for_feed_id in daily_farmgrown_feed_fed_emissions_and_resources.items():
```

Comment:
similar refactor as in the similar block in `_calculate_daily_farmgrown_feed_emissions_and_resources()`. Again, not totally necessary here but a good thing to clean up the code somewhere in the future.